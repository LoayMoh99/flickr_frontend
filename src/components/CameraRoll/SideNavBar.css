.sidenavbar {
  display: flex;
  width: 40%;
  flex-direction: column;
}

.sidenavbar a {
  display: inline-block;
}
