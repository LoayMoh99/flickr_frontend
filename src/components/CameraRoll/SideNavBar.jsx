/* eslint-disable linebreak-style */
import React from 'react';
import './SideNavBar.css';

/** Renders SideNavBar component in CameraRoll 
 * @author Khadija Khaled
 * @namespace SideNavBar
 * @category Functional Component
 * @extends Component
 */
function SideNavBar() {
  return (

    <>
      <div className= "sidenavbar">
        <a className="nav-link active " aria-current="page" href="/#">2021</a>
        <a className="nav-link " href="/#">- April</a>
        <a className="nav-link text-secondary" href="/#">- January</a>
      </div>
    </>
  );
}

export default SideNavBar;
