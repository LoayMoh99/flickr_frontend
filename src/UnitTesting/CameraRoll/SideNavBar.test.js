import React from 'react';
import ReactDOM from 'react-dom';
import SideNavBar from '../../components/CameraRoll/SideNavBar';
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('SideNavBar renders without crashing', ()=>{
    const div = document.createElement('div');
    ReactDOM.render(<SideNavBar/>, div);

});

it('SideNavBar renders correctly enzyme', () => {
    const wrapper = shallow(<SideNavBar/>)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });