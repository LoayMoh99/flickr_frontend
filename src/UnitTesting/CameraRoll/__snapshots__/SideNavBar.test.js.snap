// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavBar renders correctly enzyme 1`] = `
<Fragment>
  <div
    className="sidenavbar"
  >
    <a
      aria-current="page"
      className="nav-link active "
      href="/#"
    >
      2021
    </a>
    <a
      className="nav-link "
      href="/#"
    >
      - April
    </a>
    <a
      className="nav-link text-secondary"
      href="/#"
    >
      - January
    </a>
  </div>
</Fragment>
`;
